### PR TITLE
Command-line only plugin install action 

### PR DESCRIPTION
**Pull request by scrawfor99**
### Description
This PR builds on the recent PR for adding a plugin install action. This version drops the docker use from the Ubuntu runners and relies entirely on a standardized process for both operating systems. 

### Testing
Passes CI and successfully completes install action on Windows and Linux with JDK 11 and JDK 17. 

### Check List
- [x] New functionality includes testing
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/actions/start-opensearch-with-one-plugin/action.yml [hunk 1]
```diff
@@ -63,37 +61,24 @@ runs:
...
-        RUN useradd -u 1001 -s /sbin/nologin opensearch
-        ENV PATH="/opensearch/bin:${PATH}"
-        WORKDIR /opensearch/
-        ENTRYPOINT /docker-host/os-ep.sh
-        EOF
+
+        chmod +x ./opensearch-${{ inputs.opensearch-version }}-SNAPSHOT/bin/opensearch-plugin
+        /bin/bash -c "yes | ./opensearch-${{ inputs.opensearch-version }}-SNAPSHOT/bin/opensearch-plugin install file:$(pwd)/${{ inputs.plugin-name }}.zip"
+        chmod +x ./opensearch-${{ inputs.opensearch-version }}-SNAPSHOT/plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
+        /bin/bash -c "yes | ./opensearch-${{ inputs.opensearch-version }}-SNAPSHOT/plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh"
```
Comment (DarshitChanpura) [inline .github/actions/start-opensearch-with-one-plugin/action.yml:68]: nice!

File: .github/actions/start-opensearch-with-one-plugin/action.yml [hunk 2]
```diff
@@ -34,19 +30,21 @@ runs:
...
+    # Download OpenSearch
     - uses: peternied/download-file@v1
       if: ${{ runner.os == 'Linux' }}
       with:
-        url: https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ inputs.opensearch-version }}/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-${{ inputs.opensearch-version }}-linux-x64.tar.gz
+        url: https://artifacts.opensearch.org/snapshots/core/opensearch/${{ inputs.opensearch-version }}-SNAPSHOT/opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-linux-x64-latest.tar.gz
 
     # Extract downloaded zip
-    - name: Extract downloaded zip for Linux
+    - name: Extract downloaded zip
```
Comment (DarshitChanpura) [inline .github/actions/start-opensearch-with-one-plugin/action.yml:40]: this step looks like it is for Linux only

**Review by DarshitChanpura — COMMENTED** [review]
Looks good @scrawfor99 !

A general comment: See if we can have a variable that is initialized with `-SNAPSHOT` or `` based on the parameter passed to the script and replace current use of `-SNAPSHOT` with that variable. Thoughts?

In `action.yml`:
```
..
..
runs:
  using: "composite"
  env:
     useSnapshot: ${{ fromJSON('{"false": "danger", "true": "good"}')[inputs.use-snapshot == 'true'] }}
     # Replace all -SNAPSHOT with this variable
..
..
```

In `plugin-install.yml`:
```
..
..
plugin-start-script: install_demo_configuration
use-snapshot: true
..
..

```
Maybe this works, maybe it doesn't. But if its takes long to make this change then its not worth it.

File: .github/actions/start-opensearch-with-one-plugin/action.yml [hunk 3]
```diff
@@ -34,19 +30,21 @@ runs:
       with:
         url: https://artifacts.opensearch.org/snapshots/core/opensearch/${{ inputs.opensearch-version }}-SNAPSHOT/opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-windows-x64-latest.zip
 
+    # Download OpenSearch
     - uses: peternied/download-file@v1
       if: ${{ runner.os == 'Linux' }}
       with:
-        url: https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ inputs.opensearch-version }}/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-${{ inputs.opensearch-version }}-linux-x64.tar.gz
+        url: https://artifacts.opensearch.org/snapshots/core/opensearch/${{ inputs.opensearch-version }}-SNAPSHOT/opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-linux-x64-latest.tar.gz
```
Comment (peternied) [inline .github/actions/start-opensearch-with-one-plugin/action.yml:37]: Why change the source?  `https://ci.opensearch.org/...` [1] is the freshest place to get them from according to the OpenSearch-Build team [2]

[1] https://github.com/opensearch-project/opensearch-build/blob/main/README.md#latest-distribution-url
[2] https://github.com/opensearch-project/opensearch-build/issues/2754

**Comment by scrawfor99** [comment]
Update: I spent today trying to incorporate the ability to define a setup.sh/.bat script in the workflow into this command-line only install action. I got it working on Linux with no docker use. The issue I am running into prevent Windows from working is described in detail here https://superuser.com/questions/1753564/how-to-pass-multiple-automatic-reponses-to-user-prompts-between-batch-files-usin. 

If anyone would like to look at Linux version which does work it can be viewed on https://github.com/scrawfor99/security/tree/SetupForPluginInstall and then by swapping the os in the workflow matrix to 'ubuntu-latest' instead of windows. I also went ahead and created two files that can be used to replicate the scenario on a local windows machine or a remote desktop. The contents of the two are pasted below (they are really short) 


test.bat: 

```
@echo off
echo                                     
echo.
echo Yes or No? (y/n)
set /p Input=Enter Yes or No:
If /I "%Input%"=="y" goto yes1
goto no1
:yes1
echo You said yes 
goto yorn1
:no1
echo You said no
goto yorn1
:yorn1
echo Yes or No? (y/n)
set /p Input=Enter Yes or No:
If /I "%Input%"=="y" goto yes2
goto no2
:yes2
echo You said yes 
goto yorn2
:no2
echo You said no
goto yorn2
:yorn2
echo Yes or No? (y/n)
set /p Input=Enter Yes or No:
If /I "%Input%"=="y" goto yes3
goto no3
:yes3
echo You said yes 
goto exitprocess
:no3
echo You said no 
goto exitprocess
:exitprocess
```


run_test.bat:
```
(echo y&&echo y&&echo n) | .\test.bat
```
It should work if you can call `.\run_test.bat` in the same directory as the test.bat file and see responses that say you said 'yes' twice and 'no' once.

File: .github/actions/start-opensearch-with-one-plugin/action.yml (hunk 2 above)
Comment (scrawfor99) [reply .github/actions/start-opensearch-with-one-plugin/action.yml:40]: Yeah I briefly flew too close to the Sun and thought I could combine the two.

File: .github/actions/start-opensearch-with-one-plugin/action.yml (hunk 3 above)
Comment (scrawfor99) [reply .github/actions/start-opensearch-with-one-plugin/action.yml:37]: So I swapped the source for two reasons: 1) We do not have the same type of catalogue for Windows as we do Linux so what was available to Linux was not for Windows which made the processes for the two more divergent then I liked 2) Related to that, the most recent distributions seemed to have security already installed which would require that I make the script uninstall and reinstall the plugin to actually test what they are meant to test. Likewise, they probably would have had other plugins installed and I am not sure how much I could account for the differences in what those install and removal processes looked like. With the minimums snapshots, there is basically a barebones framework of OpenSearch to work with which I think makes it easier to install into but also gives you a better idea of whether your particular plugin is working right. 

Good question though--I actually thought a bit about this.

**Comment by scrawfor99** [comment]
> 

For the SNAPSHOT, I think that it is always going to be there so long as we use the distributions I selected. If you check out the response I sent to Peter, there were a couple reasons I chose to make both Linux and Windows runners use the min distributions. I am not sure what you guys will think but for now I think they should always have SNAPSHOT as a suffix. I can swap to your suggestion if that changes though!

**Comment by scrawfor99** [comment]
> Looks good @scrawfor99 !
> 
> A general comment: See if we can have a variable that is initialized with `-SNAPSHOT` or `<empty-string>` based on the parameter passed to the script and replace current use of `-SNAPSHOT` with that variable. Thoughts?
> 
> In `action.yml`:
> 
> ```
> ..
> ..
> runs:
>   using: "composite"
>   env:
>      useSnapshot: ${{ fromJSON('{"false": "", "true": "-SNAPSHOT"}')[inputs.use-snapshot == 'true'] }}
>      # Replace all -SNAPSHOT with this variable
> ..
> ..
> ```
> 
> In `plugin-install.yml`:
> 
> ```
> ..
> ..
> plugin-start-script: install_demo_configuration
> use-snapshot: true
> ..
> ..
> ```
> 
> Maybe this works, maybe it doesn't. But if its takes long to make this change then its not worth it.

For the SNAPSHOT, I think that it is always going to be there so long as we use the distributions I selected. If you check out the response I sent to Peter, there were a couple reasons I chose to make both Linux and Windows runners use the min distributions. I am not sure what you guys will think but for now I think they should always have SNAPSHOT as a suffix. I can swap to your suggestion if that changes though!

File: .github/actions/start-opensearch-with-one-plugin/action.yml (hunk 3 above)
Comment (peternied) [reply .github/actions/start-opensearch-with-one-plugin/action.yml:37]: > the most recent distributions seemed to have security already installed

That doesn't sound right, the `opensearch-min-*` is suppose to be only the core version of OpenSearch with no plugins, could you see about making a bug on OpenSearch-Build if that is the case.

**Comment by DarshitChanpura** [comment]
> Thank you for your review @DarshitChanpura!
> 
> For the SNAPSHOT, I think that it is always going to be there so long as we use the distributions I selected. If you check out the response I sent to Peter, there were a couple reasons I chose to make both Linux and Windows runners use the min distributions. I am not sure what you guys will think but for now I think they should always have SNAPSHOT as a suffix. I can swap to your suggestion if that changes though!

yes having SNAPSHOT is absolutely ok. My question is more in terms of defining a boolean variable that can be passed dynamically and based on that the script decides whether to append `-SNAPSHOT` after the version

**Comment by scrawfor99** [comment]
Closing in favor of #2271

**Pull request closed without merging**